cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -191,6 +193,7 @@ public MasterServiceMain() {
     this.logAppenderInitializer = injector.getInstance(LogAppenderInitializer.class);
     this.zkClient = injector.getInstance(ZKClientService.class);
     this.leaderElection = createLeaderElection();
+    this.fileContext = injector.getInstance(FileContext.class);
```

Comment:
We shouldn't expose the `FileContext`. Why we have to the `FileContext` directly instead of using `LocationFactory`?